source/Octopus.Server.Client/Repositories/Async/ReleaseRepository.cs

```diff
@@ -1,32 +1,64 @@
+
+        /// <param name="release"></param>
+        /// <param name="skip">Number of records to skip</param>
+        /// <param name="take">Number of records to take (First supported in Server 3.14.15)</param>
+        /// <param name="cancellationToken">Request cancellation token</param>
+        Task<ResourceCollection<DeploymentResource>> GetDeployments(ReleaseResource release, int skip, int? take, CancellationToken cancellationToken);
+
+
         /// <param name="release"></param>
         /// <param name="skip">Number of records to skip</param>
         /// <param name="take">Number of records to take (First supported in Server 3.14.15)</param>
-        /// <returns></returns>
+        [Obsolete("Please use the overload with cancellation token instead.", false)]
         Task<ResourceCollection<ArtifactResource>> GetArtifacts(ReleaseResource release, int skip = 0, int? take = null);
+
+        /// <param name="release"></param>
+        /// <param name="cancellationToken">Request cancellation token</param>
+        Task<ResourceCollection<ArtifactResource>> GetArtifacts(ReleaseResource release, CancellationToken cancellationToken);
+
+        /// <param name="release"></param>
+        /// <param name="skip">Number of records to skip</param>
+        /// <param name="take">Number of records to take (First supported in Server 3.14.15)</param>
+        /// <param name="cancellationToken">Request cancellation token</param>
+        Task<ResourceCollection<ArtifactResource>> GetArtifacts(ReleaseResource release, int skip, int? take, CancellationToken cancellationToken);
+
+        [Obsolete("Please use the overload with cancellation token instead.", false)]
         Task<DeploymentTemplateResource> GetTemplate(ReleaseResource release);
+        Task<DeploymentTemplateResource> GetTemplate(ReleaseResource release, CancellationToken cancellationToken);
+        
+        [Obsolete("Please use the overload with cancellation token instead.", false)]
         Task<DeploymentPreviewResource> GetPreview(DeploymentPromotionTarget promotionTarget);
-        Task<ReleaseResource> SnapshotVariables(ReleaseResource release);    
+        Task<DeploymentPreviewResource> GetPreview(DeploymentPromotionTarget promotionTarget, CancellationToken cancellationToken);
+        
+        [Obsolete("Please use the overload with cancellation token instead.", false)]
+        Task<ReleaseResource> SnapshotVariables(ReleaseResource release);
+        Task<ReleaseResource> SnapshotVariables(ReleaseResource release, CancellationToken cancellationToken);    
+        
+        [Obsolete("Please use the overload with cancellation token instead.", false)]
         Task<ReleaseResource> Create(ReleaseResource release, bool ignoreChannelRules = false);
+        Task<ReleaseResource> Create(ReleaseResource release, bool ignoreChannelRules, CancellationToken cancellationToken);
```

Comment:
Following existing pattern to handle optional param on original contract, this is missing: `Create(ReleaseResource, CancellationToken)` overload. However, this is not possible as the signature is clashing with `ICreate<TResource>.Create(TResource, CancellationToken)`, which is a `BasicRepository<>`  ([here](https://github.com/OctopusDeploy/OctopusClients/pull/681/files#diff-d471751439d55055de96cbb068826c969e2038940755f541079cf76e84c583a3L32))